exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/fn/JsonReader.java

```diff
@@ -93,13 +93,15 @@ public void ensureAtLeastOneField(ComplexWriter writer) {
     if (!atLeastOneWrite) {
       // if we had no columns, create one empty one so we can return some data for count purposes.
       SchemaPath sp = columns.get(0);
-      PathSegment root = sp.getRootSegment();
+      PathSegment fieldPath = sp.getRootSegment();
       BaseWriter.MapWriter fieldWriter = writer.rootAsMap();
-      while (root.getChild() != null && !root.getChild().isArray()) {
-        fieldWriter = fieldWriter.map(root.getNameSegment().getPath());
-        root = root.getChild();
+      while (fieldPath.getChild() != null && ! fieldPath.getChild().isArray()) {
+        fieldWriter = fieldWriter.map(fieldPath.getNameSegment().getPath());
+        fieldPath = fieldPath.getChild();
+      }
+      if (fieldWriter.isEmptyMap()) {
```

Comment:
That should be just a renaming:

It looked like the variable originally named "root" related to the root (from getRootSegment()) only until it was modified in the loop (so the name was misleading from the beginning of the loop and especially at the last reference).

Renaming it with "field" was to reflect that it corresponds to variable "fieldWriter" even as both are possibly modified in the loop.